client/pages/trip-todo.jsx

```diff
@@ -109,19 +110,46 @@ export default class TripTodo extends React.Component {
       .catch(err => console.error(err));
   }
 
+  deleteTodo(todoId) {
+    const newTodos = [];
+    for (let i = 0; i < this.state.todos.length; i++) {
```

Comment:
You only need to copy the array of you don't plan on modifying any of the objects inside.